### ISSUE TITLE
Postgres adapter's `insert_pk` is `nil` if the model's `from` is a `Sequel::SQL::DelayedEvaluation`.

### ISSUE THREAD

**Issue by tomasmiguez** [jeremyevans/sequel#2230]
### Complete Description of Issue

When trying to do an `import` on a model while setting the `return` option in `primary_key`, the `insert_pk` method gets called, but the if the model was instantiated with a `DelayedEvaluation` as source (we need this as the schema changes between requests and is read from a thread variable when executing each query) the `insert_pk` is nil as this case doesn't consider the possibility of `t` being a `DelayedEvaluation`:
`sequel/adapters/shared/postgres.rb:2430`
```ruby
      # Return the primary key to use for RETURNING in an INSERT statement
      def insert_pk
        (f = opts[:from]) && !f.empty? && (t = f.first)
        case t
        when Symbol, String, SQL::Identifier, SQL::QualifiedIdentifier
          if pk = db.primary_key(t)
            Sequel::SQL::Identifier.new(pk)
          end
        end
      end
```

One possible solution would be contemplating this new class, as this is used to execute a query and not only build it, we can  safely execute the delayed evaluation without causing issues, so something like this would work:

```ruby
      # Return the primary key to use for RETURNING in an INSERT statement
      def insert_pk
        (f = opts[:from]) && !f.empty? && (t = f.first)

        pk = case t
             when Sequel::SQL::DelayedEvaluation
               db.primary_key(t.call(self))
             when Symbol, String, SQL::Identifier, SQL::QualifiedIdentifier
               db.primary_key(t)
             end

        Sequel::SQL::Identifier.new(pk) if pk
      end
```

Would implementing something like this be an option? Is there any other more general solution possible? 

If this approach is acceptable, I wouldn't mind creating a PR and adding tests/docs as needed.

### Ruby Version

3.3.5

### Sequel Version

5.84.0

**Comment by tomasmiguez** [comment]
Or maybe something like
```ruby
      # Return the primary key to use for RETURNING in an INSERT statement
      def insert_pk
        (f = opts[:from]) && !f.empty? && (t = f.first)

        t = t.call(self) if t.is_a? Sequel::SQL::DelayedEvaluation

        case t
        when Symbol, String, SQL::Identifier, SQL::QualifiedIdentifier
          if pk = db.primary_key(t)
            Sequel::SQL::Identifier.new(pk)
          end
        end
      end
```
to be less disruptive with the existing code.

**Comment by jeremyevans** [comment]
Thanks you for the report. I think your proposal is good.  Can you please submit a pull request for it, including updates to the specs?

**Comment by tomasmiguez** [comment]
Done! https://github.com/jeremyevans/sequel/pull/2232

**Comment by jeremyevans** [comment]
Fixed by 67de9f22c21a4b40e5f856bfcb8315582f5c2a93
